### ISSUE TITLE
Token Error During Portal Upgrade

### ISSUE THREAD

**Issue by kphaneuf** [Esri/arcgis-powershell-dsc#229]
Both times we have tried to upgrade with PowerShell DSC, during the Portal upgrade we get a token error.  The most recent time when I ran the process again it worked.  But I'm not sure why we are getting the initial error.  The PSA login and password are correct.  I checked before running the upgrade.  I've attached the logs from today and the last time we tried in December.

We have 2 machines, one with Portal, Server, 2 WA's, and the other machine with Data Store.  We are upgrading from 10.6.1 to 10.7.1.  The Portal/Server machine is Windows 2012R2, the Data Store Machine is Windows 2016.

[PortalUpgrade-1-10-2020-1-07-38-PM-Verbose.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4048458/PortalUpgrade-1-10-2020-1-07-38-PM-Verbose.txt)
[PortalUpgrade-1-10-2020-10-40-23-AM-Verbose.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4048459/PortalUpgrade-1-10-2020-10-40-23-AM-Verbose.txt)
[PortalUpgrade-1-10-2020-10-40-23-AM-Error.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4048460/PortalUpgrade-1-10-2020-10-40-23-AM-Error.txt)
[PortalUpgrade-12-6-2019-8-40-09-PM-Verbose.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4048465/PortalUpgrade-12-6-2019-8-40-09-PM-Verbose.txt)
[PortalUpgrade-12-6-2019-4-54-12-PM-Error.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4048466/PortalUpgrade-12-6-2019-4-54-12-PM-Error.txt)
[PortalUpgrade-12-6-2019-4-54-12-PM-Verbose.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4048467/PortalUpgrade-12-6-2019-4-54-12-PM-Verbose.txt)

**Comment by cameronkroeker** [comment]
@kphaneuf 

This looks similar to:

https://github.com/Esri/arcgis-powershell-dsc/issues/210

Though I am afraid we did not ever figure out a resolution. What state is the ArcGIS Enterprise portal in now? For example, if you browse to https://portal.domain.com:7443/arcgis/portaladmin from the machine portal resides on does it still indicate that its in upgrade mode?

Are there any logs within C:\arcgisportal\logs?

**Comment by kphaneuf** [comment]
These are all the logs from today.  It looks like it's upgraded.
![image](https://user-images.githubusercontent.com/13141791/72192574-1c9a5a00-33bb-11ea-81f0-c10befd9a716.png)


[portal-20200110.121712-15640-0.0.log](https://github.com/Esri/arcgis-powershell-dsc/files/4048538/portal-20200110.121712-15640-0.0.log)
[portal-20200110.083549-1660-0.0.log](https://github.com/Esri/arcgis-powershell-dsc/files/4048539/portal-20200110.083549-1660-0.0.log)
[portal-20200110.100338-1708-0.0.log](https://github.com/Esri/arcgis-powershell-dsc/files/4048540/portal-20200110.100338-1708-0.0.log)
[portal-20200110.121513-3288-0.0.log](https://github.com/Esri/arcgis-powershell-dsc/files/4048541/portal-20200110.121513-3288-0.0.log)
[portal-20200110.121735-16092-0.0.log](https://github.com/Esri/arcgis-powershell-dsc/files/4048542/portal-20200110.121735-16092-0.0.log)

**Comment by kphaneuf** [comment]
Do you know if this is a bug or something that can be fixed?

**Comment by cameronkroeker** [comment]
Thanks @kphaneuf for providing the additional logs.

The good news is based on the log entries in portal-20200110.121735-16092-0.0.log, the Portal upgrade was successful.

```
<Msg time="2020-01-10T12:25:30,662" type="INFO" code="209043" source="Portal Admin" process="14048" thread="1" methodName="" machine="DWGISAP01.CITYOFSACRAMENTO.ORG" user="" elapsed="" requestID="">The portal has been upgraded from 10.6.1 to 10.7.1 successfully.</Msg>

<Msg time="2020-01-10T14:18:14,726" type="INFO" code="209085" source="Portal Admin" process="2940" thread="1" methodName="" machine="DWGISAP01.CITYOFSACRAMENTO.ORG" user="" elapsed="" requestID="">Post Upgrade completed.</Msg>
```

We are still unsure about what could be causing the initial portal token error, however we are actively investigating it.

**Comment by kphaneuf** [comment]
Today we rolled back again and I found a few things that I think were causing this issue.  1. We use SAML to sign in and had built-in user logins turned off on the sign in page.  I changed this and signed in as the admin we used in the powershell config file.  This user had never logged in and had to setup the security question.  I filled that out and then added that to the powershell config file as it wasn't previously there.  I then changed it back to not showing the built in user on the sign in page.  2. the hostname.properties file did not exist in C:\Program Files\ArcGIS\Portal\framework\etc in our 10.6.1 environment.  I had saved the hostname.properties file from the 10.7.1 version and once I rolled back I put that file in the same location.  This got rid of the error 
"Configured hostname '' does not match expected value 'DWGISAP01.cityofsacramento.org'
[DWGISAP01]: LCM:  [ End    Test     ]  [[ArcGIS_PortalUpgrade]PortalUpgrade]  in 0.5470 seconds.

I also edited the hostidentifier.properties file as previously in 10.6.1 it was all commented out.  I changed the 10.6.1 file to match what was in the 10.7.1 file which was DWGISAP01.cityofsacramento.org and the preferred identifier = hostname.  For some reason during the upgrade it gave the same error 
"In Portal DataStore Configured host identifier '' does not match expected value 'DWGISAP01.cityofsacramento.org' or host identifier type '' does not match expected value 'hostname'. Setting it"

After these errors the service stopped restarted and eventually was able to connect and get a token.  I think the security questions is what was causing the issue.
